crates/context_aware_config/src/api/dimension/helpers.rs

```diff
@@ -0,0 +1,29 @@
+use serde_json::{Map, Value};
+use service_utils::service::types::Tenant;
+
+pub fn get_mandatory_dimensions(
+    mandatory_dimensions: &Map<String, Value>,
+    tenant: Tenant,
+) -> Vec<String> {
+    let result: Vec<String> = mandatory_dimensions
+        .get(&tenant.0.to_string())
+        .and_then(|v| v.as_array())
+        .unwrap_or(&vec![])
```

Comment:
shouldn't this be better
```suggestion
        .map_or_else(|| vec![], |arr| {
            arr.iter().filter_map(|value| value.as_str()).map(String::from).collect()
        |)
```